### PR TITLE
Upgrading to node 16

### PR DESCRIPTION
**Pull request by hokei**
[Jira Ticket](http://jira.nypl.org/browse/SFR-XXX)

### This PR does the following:
- Upgrades DRB to use Node 16
- Removes V1 webreader 
- Removes all dependencies on Enzyme.  
- Uses `next-router-mock` instead of our custom mock script, which has better features.

### Testing requirements & instructions: 
-

### CODE REVIEW

File: src/components/Feedback/Feedback.test.tsx [hunk 1]
```diff
@@ -56,21 +56,23 @@ describe("Feedback", () => {
...
+    window.fetch = fakeFetch;
     test("should invoke method when success and feedback set", () => {
       jest.spyOn(window, "alert").mockImplementation(() => jest.fn());
 
       const feedback = screen.getByRole("textbox", {
         name: "Comments (Required)",
       });
       const yesRadio = screen.getByLabelText("Yes");
-
+      expect(true).toBeTruthy;
```
Comment (samanthaandrews) [inline src/components/Feedback/Feedback.test.tsx:68]: stray assertion i think?

Comment (samanthaandrews) [inline src/components/Feedback/Feedback.test.tsx:68]: stray assertion i think?

File: src/__tests__/Search.test.tsx [hunk 2]
```diff
@@ -19,6 +15,10 @@ import { FilterLanguagesCommonTests } from "./componentHelpers/FilterLanguages";
 import { FilterFormatTests } from "./componentHelpers/FilterFormats";
 import { findFiltersForField } from "../util/SearchQueryUtils";
 import filterFields from "../constants/filters";
+import mockRouter from "next-router-mock";
+
+jest.mock("next/router", () => require("next-router-mock"));
```
Comment (samanthaandrews) [inline src/__tests__/Search.test.tsx:20]: is this how we are mocking the next/router? just curious; wasn't sure what this line was doing but seems to be working

**Review by samanthaandrews — CHANGES_REQUESTED** [review]
Just one small comment and then we are good to approve! I pulled in the branch locally to test that I could npm install, build the app, and run tests. Everything is working like a charm 💫

File: src/__tests__/Search.test.tsx (hunk 2 above)
Comment (hokei) [reply src/__tests__/Search.test.tsx:20]: Yes - we used to use `MockNextRouter.tsx` but it was hacked together, so using an actual library seemed more maintainable

**Review by samanthaandrews — APPROVED** [review]

**Pull request merged**